draft-ietf-taps-transport-security.md

```diff
@@ -844,6 +855,70 @@ and the per-message nonce in the clear. Everything else is encrypted.
 
 - An unreliable transport protocol such as UDP.
 
+## OpenVPN
+
+OpenVPN {{OpenVPN}} is a commonly used protocol designed to replace IPsec. A
+major goal of this protocol is to provide an easy-to-use VPN. OpenVPN
+encapsulates either IP packets or Ethernet frames within a secure tunnel and
+can run over UDP or TCP.
+
+### Protocol Description
+
+OpenVPN facilitates authentication using either a pre-shared static key or
+using X.509 certificates and TLS. In pre-shared key mode, OpenVPN derives
+keys for encryption and authentication directly from one or multiple static
+keys. In TLS mode, OpenVPN encapsulates a TLS handshake, in which both peers
+must present a certificate for authentication. After the handshake, both sides
+contribute random source material to derive keys for encryption and
+authentication using the TLS pseudo random function (PRF). OpenVPN provides the
+possibility to authenticate and encrypt the TLS handshake itself using a
+pre-shared key or passphrase. Furthermore, it supports rekeying using TLS.
+
+After authentication and key exchange, OpenVPN encrypts payload data, i.e., IP
+packets or Ethernet frames, and signs the payload using an HMAC function. The
+default cipher is BlowFish and the default message digest algorithm is SHA1,
+but an application can select an arbitrary cipher, key size, and message digest
+algorithm for the HMAC. OpenVPN peers may also support cipher negotiation
+(NCP). If both hosts allow, they can upgrade the cipher to AES-256-GCM.
```

Comment:
I would drop the "If both hosts.." sentence, as it seems implied based negotiation support.

Reply:
I'm not sure we should just drop this part. The cipher upgrade seems important to me, as BF-CBC (the default cipher) is no longer recommended.

I looked into the details again, and OpenVPN supports cipher negotiation (NCP) since version 2.4 of the software (most recent major release). So if both peers are v2.4+, in client/server mode (instead of "point-to-point"), and do not explicitly disable NCP, the client will advertise NCP support, the server will choose a cipher (by default AES-256-GCM), and tell the client to use this cipher. 
If one side is an older version, apparently there's some limited negotiation support as well, where the 2.4 side changes its cipher to the one that the other side has set with --cipher. They do not specify how exactly this limited negotiation works.

So I suggest to change this part to say:
"OpenVPN peers support cipher negotiation (NCP) since version 2.4, in which case they will upgrade the cipher to AES-256-GCM by default."

Does that work?

Reply:
Yes! Your proposal reads much better.